Embeddable serialized state

nreese opened the pull request.

Commit:
initial set up of new embeddable factory types.

Commit:
First pass of state manager. Messy version of state & unsaved changes for eui markdown

Heenawter committed:
[Embeddables][Serialized State Only] Use serialized state in state transfer service (#216232)

Fixes https://github.com/elastic/kibana/issues/213153

## Summary

This PR ensures that the embeddable state transfer service
**exclusively** uases serialized state. Previously, the transfer service
accepted **either** runtime state or serialized state - but we are in
the process of removing the concept of "runtime state" for
embeddables (at least as far as the Dashboard is concerned), and this is
one step towards that goal.

Note that this PR is built off the subset of "embeddable state transfer
service related" changes from
https://github.com/elastic/kibana/pull/215416 - all I did was clean up
these changes, finalized some of the cases that were missed in the
original draft, and updated tests as necessary.

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Commit:
Dashboard integration in progress.

Commit:
Merge remote-tracking branch 'refs/remotes/upstream/embeddable-serialized-state' into embeddable-serialized-state

Commit:
SerializedStateOnly Dashboard integration

Commit:
Use new state manager type for titles and markdown. set up unsaved changes initializer.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nreese committed:
convert search example embeddable (#217246)

nreese committed:
Migrate Map embeddable to serialized state only (#217351)

nreese committed:
Migrate Visualize embeddable to serialized state only (#217736)

Also updates state manager API

nreese committed:
Fix unlink from library (#217743)

nreese committed:
[canvas] migrate canvas to serialized state only (#217874)

nreese committed:
convert Lens embeddable to serialized state only (#217911)

nreese committed:
Discover session serialized state (#218015)

nreese committed:
Tslint serialize state (#218026)

nreese committed:
fix reading properties of undefined when currentChildState[uuid] is not provided (#218186)

Commit:
Merge remote-tracking branch 'upstream/main' into embeddable-serialized-state

nreese committed:
update titleManger to implement PublishesTitle and PublishesWritableTitle (#218369)

nickpeihl committed:
[Serialized State Only] Field list embeddable (#218174)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Updates the Field list example embeddable to serialized state only.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

nickpeihl committed:
[Embeddable] [Serialized State Only] Convert SLO embeddable to serialized state only (#218270)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

Converts the SLO embeddable to serialized state only.

## Testing this PR

Create an SLO using the "How to Test" section in the description of
[this PR](https://github.com/elastic/kibana/pull/179147).

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

nickpeihl committed:
[Embeddable] SLO Alerts embeddable to serialized state only (#218349)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

Converts the SLO alerts embeddable to serialized state only.

## Testing this PR

Create an SLO using the "How to Test" section in the description of
[this PR](https://github.com/elastic/kibana/pull/179147).

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

nickpeihl committed:
[Serialized State Only] [SLO] Convert SLO Burn rate to serialized state only (#218461)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Convert SLO burn rate to serialized state only

## Testing this PR

Create an SLO using the "How to Test" section in the description of
[this PR](https://github.com/elastic/kibana/pull/179147).

Heenawter committed:
[Serialized State Only] Data table embeddable (#218355)

## Summary

This PR removes runtime state from the example data table embeddable

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

elasticmachine committed:
Merge branch 'main' into embeddable-serialized-state

elasticmachine committed:
Merge branch 'main' into embeddable-serialized-state

nreese committed:
Convert control group embeddable to serialized state only (#218370)

elasticmachine committed:
Merge branch 'main' into embeddable-serialized-state

nreese committed:
convert traces embeddable to serialized state only (#218667)

nickpeihl committed:
[Serialized State Only] [SLO] Convert SLO Error Budget embeddable to serialized state only (#218464)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Convert SLO Error Budget embeddable to serialized state only

## Testing this PR

Create an SLO using the "How to Test" section in the description of
[this PR](https://github.com/elastic/kibana/pull/179147).

nickpeihl committed:
[Serialized State Only] Alerts embeddable serialized state only (#218684)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Converts the Alerts embeddable table to serialized state only

This embeddable is not in use yet. Testing requires uncommenting [this
line](https://github.com/elastic/kibana/blob/f1eb019b7b5d73a1e8322cdc0b0dc0fc973f9300/x-pack/platform/plugins/shared/embeddable_alerts_table/public/plugin.ts#L36)
in the embeddable alerts table plugin.

cqliu1 committed:
[Serialized State Only] Book embeddable (#218343)

## Summary

This removes runtime state from the book example embeddable.

*Note: This PR does not need to be reviewed by external teams, only
presentation team. This PR merges into a feature branch that Kibana
presentation team is working on to convert the embeddable framework to
only expose serialized state. Your team will be pinged for review once
the work is complete and the [final
PR](https://github.com/elastic/kibana/pull/217239) opens that merges the
feature branch into main

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

elasticmachine committed:
Merge branch 'main' into embeddable-serialized-state

nickpeihl committed:
[Embeddable] [Serialized State Only] Convert Synthetics stats embeddable to serialized state only (#218730)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Convert Synthetics stats embeddable to serialized state only

nreese committed:
tslint lens and maps plugins (#218732)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

nickpeihl committed:
[Serialized state only] Convert synthetic monitors embeddable to serialized state only (#218735)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Convert synthetic monitors embeddable to serialized state only

nreese committed:
copy to dashboard action (#218737)

nickpeihl committed:
[Serialized State Only] Convert APM alerting embeddables to serialized state only (#218745)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Convert APM alerting embeddables to serialized state only

nickpeihl committed:
[Serialized State Only] Convert pattern anslysis embeddable to serialized state only (#218812)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._


## Summary

Convert pattern anslysis embeddable to serialized state only

elasticmachine committed:
Merge branch 'main' into embeddable-serialized-state

Commit:
[Serialized State Only] Field Stats (#218751)

Converts the Field Stats Embeddable to Serialized State only.

olapawlus committed:
[Serialized State Only] Button embeddable (#218658)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

This removes runtime state from the button example embeddable.

nreese committed:
Merge branch 'main' into embeddable-serialized-state

nreese committed:
Dashboard locator serialized state (#218832)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

nreese committed:
Merge branch 'main' into embeddable-serialized-state

nickpeihl committed:
[Serialized State Only] Convert log rate analysis embeddable to serialized state only (#218981)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

## Summary

Convert log rate analysis embeddable to serialized state only

nreese committed:
tslint embeddable_examples plugin (#219009)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

nreese committed:
convert links embeddable to serialized state only (#219023)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

elasticmachine committed:
Merge branch 'main' into embeddable-serialized-state

nreese committed:
Convert change point embeddable to expose serialized state only (#219120)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

nreese committed:
tslint discover plugin (#219151)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

nreese committed:
tslint portable dashboard examples (#219153)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

nickpeihl committed:
[Serialized state only] Convert log stream embeddable to serialized state only (#219156)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

Convert log stream embeddable to serialized state only

## Testing this PR
Log stream embeddables are deprecated and can no longer be added to
dashboards. You'll have to import an existing dashboards saved object
that contains a log stream embeddable, such as [this
one](https://gist.github.com/nickpeihl/be15dc0645fedd9765f26a19dad6dade).

nreese committed:
tslint ml plugin (#219159)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

nreese committed:
tslint image plugin (#219163)

This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'
